scripts/updateOrderPortal.py

```diff
@@ -1,41 +1,111 @@
-""" Script to update NGI project names and ids from LIMS
+""" Script to update projects from LIMS
     to corresponding orders in Order Portal."""
 from genologics.lims import *
 from genologics.config import BASEURI, USERNAME, PASSWORD
+from genologics_sql import queries
+from genologics_sql.utils import *
 from datetime import *
 import json
 import argparse
 import requests
+import os
+import LIMS2DB.utils as lutils
 
-def main(args):
-    with open(args.config) as config_file:
-        creds = json.load(config_file)
-    OP_BASE_URL = creds['OrderPortal'].get('URL')  # Base URL for your OrderPortal instance.
-    API_KEY = creds['OrderPortal'].get('API_KEY')  # API key for the user account.
-    headers = {'X-OrderPortal-API-key': API_KEY}
+class Order_Portal_APIs(object):
+
+    def __init__(self, url, headers, log):
+
+        self.base_url = url
+        self.headers = headers
+        self.log = log
+        self.lims = Lims(BASEURI, USERNAME, PASSWORD)
+
+    def update_order_internal_id(self, open_date, dry_run):
+        pjs = self.lims.get_projects(open_date=open_date.strftime("%Y-%m-%d"))
+        for project in pjs:
+            if project.open_date:
+                project_ngi_identifier=project.id
+                project_ngi_name=project.name
+                try:
+                    ORDER_ID=project.udf['Portal ID']
+                except KeyError:
+                    continue
+
+                url = "{base}/api/v1/order/{id}".format(base=self.base_url, id=ORDER_ID)
+                data = {'fields': {'project_ngi_name': project.name, 'project_ngi_identifier': project.id}}
+                if not dry_run:
+                    response = requests.post(url, headers=self.headers, json=data)
+                    assert response.status_code == 200, (response.status_code, response.reason)
 
-    lims = Lims(BASEURI, USERNAME, PASSWORD)
-    yesterday=date.today()-timedelta(days=1)
-    pjs=lims.get_projects(open_date=yesterday.strftime("%Y-%m-%d"))
-    for project in pjs:
-        if project.open_date:
-            project_ngi_identifier=project.id
-            project_ngi_name=project.name
+                    self.log.info('Updated internal id for order:{} - {}'.format(ORDER_ID, project.id))
+                else:
+                    print('{} Dry run: Updated internal id for order:{} - {}'.format(ORDER_ID, project.id))
+
+    def update_order_status(self, to_date, dry_run):
+        lims_db = get_session()
```

Comment:
I'm not a big fan of `from x import *`, because I can't really tell where the functions are coming from, like in this place. But I'll leave that up to you to decide what you like best 🙂

Reply:
Changed it somewhat.